### PR TITLE
[security] `az security alerts-suppression-rule`: Add alerts suppression rules to security module

### PR DESCRIPTION
**Pull request by nitsi**
**Description**
Adding new functionality, we added alerts-suppression-rule commands to the security module.
The swagger can be found (here)[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/alertsSuppressionRules.json]

Alert suppression rules are used to automatically dismiss alerts based on provided rules, the API supports update (this also creates the rule), get and delete.

rule name - provided by the customer
alert type - the alert type to automatically dismiss
reason - the reason the customer is suppression this alert
scope - when suppressing alerts by alert type only do so if this scope is matched (variable name all-of same as in the swagger and python)

**Testing Guide**
```
security alerts-suppression-rule list
```
- List all suppression rules on the subscription

```
security alerts-suppression-rule show  --rule-name RuleName
```
- Get a specific rule by its name

```
az security alerts-suppression-rule update --rule-name RuleName --alert-type "Test" --reason "Other" --comment "Test comment" --state "Enabled"
```
- Create a suppression rule with name RuleName that suppress alerts when: alert type is "Test" and its DNS domain entity contains the string "testContains"

```
az security alerts-suppression-rule delete --rule-name RuleName
```
- Delete a specific rule by its name

```
 az security alerts-suppression-rule upsert_scope --field "entities.process.commandline" --containsSubstring "example" --rule-name RuleName
```
- Upsert a specific scope to an alert suppression rule

```
 az security alerts-suppression-rule delete_scope --field "entities.process.commandline" --rule-name RuleName
```
- Delete a specific scope in an alert suppression rule

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by kairu-ms** [comment]
Please fix CI issues as well

File: src/azure-cli/azure/cli/command_modules/security/_params.py [hunk 1]
```diff
@@ -25,6 +25,16 @@
...
+suppression_alert_type_arg_type = CLIArgumentType(options_list=('--alert-type'), metavar='ALERTTYPE', help='Type of the alert to automatically suppress. For all alert types, use "*".')
+suppression_reason_arg_type = CLIArgumentType(options_list=('--reason'), metavar='REASON', help='The reason for dismissing the alert.')
+suppression_expiration_date_utc_arg_type = CLIArgumentType(options_list=('--expiration-date-utc'), metavar='EXPIRATIONDATEUTC', help='Expiration date of the rule, if value is not provided or provided as null this field will default to the maximum allowed expiration date.')
+suppression_state_arg_type = CLIArgumentType(options_list=('--state'), metavar='STATE', help='Possible states of the rule. Possible values are "Enabled" and "Disabled".')
+suppression_comment_arg_type = CLIArgumentType(options_list=('--comment'), metavar='COMMENT', help='Any comment regarding the rule.')
+suppression_all_of_arg_type = CLIArgumentType(options_list=('--all-of'), metavar='ALLOF', help='The suppression conditions. Should be provided in a json array format.')
```
[the remark this answers is not in the document]
Comment (nitsi) [reply src/azure-cli/azure/cli/command_modules/security/_params.py:36]: I followed the guideline [here](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md) to handle an array

**Comment by nitsi** [comment]
@kairu-ms Hi, can you please re-run `Azure.azure-cli (Check the Format of Pull Request Title and Content)` I updated the title but I don't have permissions to rerun the pipeline

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**